game.py

```diff
@@ -25,14 +25,21 @@
 
 class Game:
 
-    def __init__(self, bg_image_path, keymap, border_width=16, speed=7):
+    def __init__(self, bg_image_path, keymap, config,
+                 border_width=16, speed=7):
         info = pygame.display.Info()
         self.background = pygame.image.load(bg_image_path)
         self.background = pygame.transform.scale(self.background,
                                                  (600, info.current_h))
+
+        self.config = config
+        self.speed_multiplier = [0.7, 1, 1.3][config["difficulty"]]
+        self.score_multiplier = [0.5, 1, 2][config["difficulty"]]
```

Comment:
There's no need to create `self.config` as this is the only place it's used, a better thing would be to create a difficulty variable that's assigned `config.get("difficulty", 1)` and use that where needed.

If 1 is the least difficulty value then I don't think we need 0.7 and 0.5 as part of the list, what made you want to use those values?

Reply:
I am planning that any setttings which can be added to the settings option that I added in the main menu will add their properties to config. So for eg. volume or anything else can be stored in config and since at that time it will be accessed many times in different places I thought I'd start by making it a class property.

Besides, the difficulty ranges from 0 to 2. and the speed_multiplier is just a constant the speed will be multiplied with for the spikes. These values seemed balanced in terms of gameplay and difficulty. Any better suggestions for these values are welcome.

Reply:
> I am planning that any setttings which can be added to the settings option that I added in the main menu will add their properties to config. So for eg. volume or anything else can be stored in config and since at that time it will be accessed many times in different places I thought I'd start by making it a class property.

Makes sense but at the moment that hasn't been implemented so it'll be better for it not to be a class variable until we get to when that's necessary.

And also the way it's been used is currently not readable.
 
> Besides, the difficulty ranges from 0 to 2. and the speed_multiplier is just a constant the speed will be multiplied with for the spikes. These values seemed balanced in terms of gameplay and difficulty. Any better suggestions for these values are welcome.

Alright, I asked as I wanted to understand your thought process, haven't tested yet to know what other values would work.